o4/o4.py

```diff
@@ -496,26 +493,36 @@ def o4_pyforce(debug, no_revision, args: list, quiet=False):
 
     from tempfile import NamedTemporaryFile
     from collections import defaultdict
+    from stat import S_IWUSR, S_IFREG
 
     class LogAndAbort(Exception):
         'Dumps debug information on errors.'
 
-    o4_log('pyforce', no_revision=no_revision, quiet=quiet, *args)
+    a_plus_w = 0o222
+    a_minus_w = (0o777 ^ a_plus_w) | S_IFREG
+
+    o4_log('pyforce', no_revision=no_revision, writable=writable, quiet=quiet, *args)
 
     tmpf = NamedTemporaryFile(dir='.o4')
     fstats = []
+    orig_mode = {}
     for line in sys.stdin.read().splitlines():
         if line.startswith('#o4pass'):
             print(line)
             continue
         f = fstat_split(line)
         if f and caseful_accurate(f[F_PATH]):
             fstats.append(f)
+            try:
+                s = os.stat(f[F_PATH])
+                if s.st_mode & S_IWUSR:
+                    orig_mode[f[F_PATH]] = s.st_mode
+                    os.chmod(f[F_PATH], s.st_mode & a_minus_w)
```

Comment:
Is this all just in case the guy (or his stupid computer) made the file writable? What does this do to elapsed time?

Reply:
Shouldn't this be done only if --writable?